src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        private readonly static Metadata Metadata = new("Flagsmith Provider");
+        internal readonly IFlagsmithClient _flagsmithClient;
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        public FlagsmithProvider(IFlagsmithConfiguration options)
+        {
+            _flagsmithClient = new FlagsmithClient(options);
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        /// <param name="httpClient">Http client that will be used for flagsmith requests. You also can use it to register <see cref="FeatureProvider"/> as Typed HttpClient with <see cref="FeatureProvider"> as abstraction</see></param>
+        public FlagsmithProvider(IFlagsmithConfiguration options, HttpClient httpClient)
+        {
+            _flagsmithClient = new FlagsmithClient(options, httpClient);
+        }
+
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="flagsmithClient">Precreated Flagsmith client. You can just use <see cref="FlagsmithClient"/> class.</param>
+        public FlagsmithProvider(IFlagsmithClient flagsmithClient)
+        {
+            _flagsmithClient = flagsmithClient;
+        }
+
+        private Task<IFlags> GetFlags(EvaluationContext ctx = null)
+        {
+            var key = ctx?.GetValue("targetingKey")?.AsString;
+            return string.IsNullOrEmpty(key)
+                ? _flagsmithClient.GetEnvironmentFlags()
+                : _flagsmithClient.GetIdentityFlags(key, ctx.AsDictionary().Select(x => new Trait(x.Key, x.Value.AsObject) as ITrait).ToList());
+        }
+
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata() => Metadata;
+
+        /// <inheritdoc/>
+
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            var flags = await GetFlags(context);
+            var isFlagEnabled = await flags.IsFeatureEnabled(flagKey);
+            if (!isFlagEnabled)
+            {
+                return new ResolutionDetails<bool>(flagKey, defaultValue, reason: Reason.Disabled);
+            }
+
+            var stringValue = await flags.GetFeatureValue(flagKey);
+            if (bool.TryParse(stringValue, out var parsedValue))
+            {
+                return new ResolutionDetails<bool>(flagKey, parsedValue);
+            }
+            return new ResolutionDetails<bool>(flagKey, defaultValue, ErrorType.ParseError, Reason.Error, errorMessage: "Failed to parse value in int type");
```

Comment:
I recommend throwing [TypeMismatch ](https://github.com/open-feature/dotnet-sdk/blob/main/src/OpenFeature/Error/TypeMismatchException.cs) instead, in which case the SDK will handle the error.

Reply:
ok. I'll change it to throwing an exception

Reply:
Can't find this exception in 0.5 version. should I update minimal version?

Reply:
Changed. But also changed version to 1.3.1 (only for this new package)

Reply:
> Can't find this exception in 0.5 version. should I update the minimal version?

Yes. I would make the min version 1.0

Reply:
done